amundsen_application/static/js/components/AnnouncementPage/index.tsx

```diff
@@ -50,32 +45,34 @@ export class AnnouncementPage extends React.Component<AnnouncementPageProps, Ann
     return (
       <div key={`post:${postIndex}`} className='post-container'>
         <div className='post-header'>
-          <div className='post-title'>{post.title}</div>
-          <div className='post-date'>{post.date}</div>
+          <div id='post-title' className='post-title'>{post.title}</div>
+          <div id='post-date' className='post-date'>{post.date}</div>
         </div>
-        <div className='post-content'>
+        <div id='post-content' className='post-content'>
```

Comment:
For repeated elements we should try to use unique IDs. Maybe `id={'post-content-${postIndex}'}`

Reply:
Also going to move away from finding elements by id, it will eventually cause headaches for sure and is not a best practice.  

I'm still playing around with what's best, we can always drill down normally like `subject.find('div').find('div').find(...)`, but that can also get overwhelming. My current thinking is that this is also why some folks break their components down into so many sub-components to make testing easier. 

I've been trying to do this without refactoring components, but starting to feel worried about introducing any bad test practices. I think I will attempt to tackle `SearchPage` and it's nested components using as many best practices as I'm aware of and see how we feel about that.